### DISCUSSION TITLE
How to compare things via a comparator?

### DISCUSSION THREAD

**Discussion by SerVB** [robstoll/atrium#1261] [Q&A] [answered]
For example, I have a comparator for Strings: `compareBy<String> { it.lowercase() }`. How can I use it to compare an expected string to an actual one?

**Comment by robstoll** [comment, marked as answer]
Atrium does not provide any function involving Comparator so far, but that can change of course :)

Right now you could do (without writting an additional extension method):
```
expect(person.firstName)
  .feature("compare against Sergei using myLowerCaseComparator") { myLowerCaseComparator(this, "Sergei") }
  .toEqual(0)
```

So a few questions, what is your exact use case? something like:
1.  ```
    expect(person.firstName)  // compare it somehow against "Sergei" using myLowerCaseComparator and the result should be 0
    ```
2. Would you like to use your comparator as well to  compare elements in Ierables as well.
3. Are you only interested in the `0` outcome or would you also like to expect on < 0 or > 0 ?

**Reply by SerVB** [reply]
We've decided to go with another assertions library... Feel free to disregard my initial question. Let me answer but probably it's not worth implementing unless a real user of atrium wants it.
1. Something like this: `expect("Sergei").toBeEqualComparedWith("sERGEI", compareBy<String> { it.lowercase() })`
2. I think I meant only the case of comparing two things given the comparator, not a deep comparison...
3. I think I meant only 0. Although, methods like `toBeLessThanComparedWith` could be helpful as well...

**Reply by robstoll** [reply]
Thanks for the additional feedback
